sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -3232,6 +3232,35 @@ def compile_and_reload(
             pipeline_func=pipeline, package_path=output_yaml)
         return components.load_component_from_file(output_yaml)
 
+class TestResourceConfig(unittest.TestCase):
+    def test_cpu_memory_optional(self):
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            predict_op = components.load_component_from_text("""
+      name: predict
+      inputs:
+      - {name: generate_explanation, type: Boolean, default: False}
+      implementation:
+        container:
+          image: gcr.io/my-project/my-image:tag
+          args:
+          - {inputValue: generate_explanation}
+      """)
+
+            @dsl.pipeline(name='test-resource-pipeline')
```

Comment:
nit: consider removing name and just using the uncalled `dsl.pipeline` decorator to minimize code not related to the behavior being tested